tests/setupAfterEnv.ts

```diff
@@ -77,4 +80,49 @@ expect.extend({
           pass: true,
         };
   },
+
+  /**
+   * Tests that the given promise is rejected within a certain amount of time
+   * (which defaults to the time that Jest tests wait before timing out as
+   * configured in the Jest configuration file).
+   *
+   * Inspired by <https://stackoverflow.com/a/68409467/260771>.
+   *
+   * @param promise - The promise to test.
+   * @returns The result of the matcher.
+   */
+  async toBeRejected(promise: Promise<any>) {
```

Comment:
Hmm, this is neat, but what advantages does this have over using `.rejects` and relying upon the built-in timeout?

I guess this would be useful in cases where the timeout is disabled or is very long (e.g. the extension e2e tests), but we're not in that situation here. Or when you want quite a short timeout just for this one step.

Reply:
Yeah, I guess instead of using this matcher I could use `await expect(...).rejects.toThrow(expect.anything())`. It's symmetrical with `toResolve`, which I like, but I guess it's not completely necessary. Okay, I can make a replacement.

Reply:
I'm ok with keeping it as well if you want! It has some utility even if we don't strictly need it here. I was just making sure I wasn't missing something

Reply:
It's okay. We can always add it back!